managed/data_source_integration.go

```diff
@@ -0,0 +1,319 @@
+					"access_policy_token": {
+						Description: "Grafana Access Policy Token",
+						Type:        types.StringType,
+						Computed:    true,
+						Sensitive:   true,
+					},
+					"zone": {
+						Description: "Grafana Zone.",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"instance_id": {
+						Description: "Grafana InstanceID.",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"org_slug": {
+						Description: "Grafana OrgSlug.",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+				}),
+			},
+			"sumologic_spec": {
+				Description: "The specifications of a Sumo Logic integration.",
+				Computed:    true,
+				Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+					"access_id": {
+						Description: "Sumo Logic Access Key ID",
+						Type:        types.StringType,
+						Computed:    true,
+						Sensitive:   true,
+					},
+					"access_key": {
+						Description: "Sumo Logic Access Key",
+						Type:        types.StringType,
+						Computed:    true,
+						Sensitive:   true,
+					},
+					"installation_token": {
+						Description: "A Sumo Logic installation token to export telemetry to Grafana with",
+						Type:        types.StringType,
+						Computed:    true,
+						Sensitive:   true,
+					},
+				}),
+			},
+			"googlecloud_spec": {
+				Description: "The specifications of a Google Cloud integration.",
+				Computed:    true,
+				Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+					"type": {
+						Description: "Service Account Type",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"project_id": {
+						Description: "GCP Project ID",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"private_key": {
+						Description: "Private Key",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"private_key_id": {
+						Description: "Private Key ID",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"client_email": {
+						Description: "Client Email",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"client_id": {
+						Description: "Client ID",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"auth_uri": {
+						Description: "Auth URI",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"token_uri": {
+						Description: "Token URI",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"auth_provider_x509_cert_url": {
+						Description: "Auth Provider X509 Cert URL",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"client_x509_cert_url": {
+						Description: "Client X509 Cert URL",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"universe_domain": {
+						Description: "Google Universe Domain",
```

Comment:
universe domain is typically an optional field in GCP spec. 
Should it be marked as Optional: true?

Reply:
Marking it as `Optional: true` allows the user to provide this value in the config if needed. However, since this is a data source, user input is not required. The field is automatically computed, and if no value is provided, it will default to an empty string (`""`).